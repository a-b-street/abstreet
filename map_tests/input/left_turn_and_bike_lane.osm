<?xml version='1.0' encoding='UTF-8'?>
<!-- Resembles https://www.openstreetmap.org/node/53086620: there's a left turn lane and a bike lane. -->
<osm>
        <bounds minlon="0.0" maxlon="0.01" minlat="0.0" maxlat="0.01"/>
        <node id="1" lon="0.005" lat="0.005"/>
        <node id="2" lon="0.005" lat="-1.0"/>
        <node id="3" lon="0.005" lat="1.0"/>
        <node id="4" lon="-0.1" lat="0.005"/>
        <node id="5" lon="1.0" lat="0.005"/>
        <way id="100">
            <nd ref="1"/>
            <nd ref="2"/>
            <tag k="name" v="south"/>
            <tag k="highway" v="primary"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="4"/>
        </way>
        <way id="101">
            <nd ref="1"/>
            <nd ref="3"/>
            <tag k="name" v="north"/>
            <tag k="highway" v="primary"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="5"/>
            <tag k="lanes:forward" v="2"/>
            <tag k="lanes:backward" v="3"/>
            <tag k="turn:lanes:backward" v="left||"/>
        </way>
        <way id="102">
            <nd ref="1"/>
            <nd ref="4"/>
            <tag k="name" v="west"/>
            <tag k="highway" v="residential"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="2"/>
        </way>
        <way id="103">
            <nd ref="1"/>
            <nd ref="5"/>
            <tag k="name" v="east"/>
            <tag k="highway" v="residential"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="2"/>
            <tag k="cycleway" v="lane"/>
        </way>
</osm>
