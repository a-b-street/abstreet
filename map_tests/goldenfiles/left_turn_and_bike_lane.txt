TurnID(Lane #0, Lane #5, Intersection #0) is a Crosswalk
TurnID(Lane #0, Lane #22, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #1, Lane #10, Intersection #0) is a Straight
TurnID(Lane #1, Lane #11, Intersection #0) is a Straight
TurnID(Lane #1, Lane #20, Intersection #0) is a Right
TurnID(Lane #1, Lane #21, Intersection #0) is a Right
TurnID(Lane #2, Lane #10, Intersection #0) is a Straight
TurnID(Lane #2, Lane #11, Intersection #0) is a Straight
TurnID(Lane #2, Lane #15, Intersection #0) is a Left
TurnID(Lane #2, Lane #21, Intersection #0) is a Right
TurnID(Lane #5, Lane #0, Intersection #0) is a Crosswalk
TurnID(Lane #5, Lane #13, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #6, Lane #12, Intersection #0) is a Crosswalk
TurnID(Lane #6, Lane #16, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #7, Lane #3, Intersection #0) is a Straight
TurnID(Lane #7, Lane #4, Intersection #0) is a Straight
TurnID(Lane #7, Lane #15, Intersection #0) is a Right
TurnID(Lane #7, Lane #21, Intersection #0) is a Left
TurnID(Lane #8, Lane #3, Intersection #0) is a Straight
TurnID(Lane #8, Lane #4, Intersection #0) is a Straight
TurnID(Lane #8, Lane #21, Intersection #0) is a Left
TurnID(Lane #9, Lane #20, Intersection #0) is a Left
TurnID(Lane #9, Lane #21, Intersection #0) is a Left
TurnID(Lane #12, Lane #6, Intersection #0) is a Crosswalk
TurnID(Lane #12, Lane #17, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #13, Lane #5, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #13, Lane #16, Intersection #0) is a Crosswalk
TurnID(Lane #14, Lane #3, Intersection #0) is a Right
TurnID(Lane #14, Lane #4, Intersection #0) is a Right
TurnID(Lane #14, Lane #10, Intersection #0) is a Left
TurnID(Lane #14, Lane #11, Intersection #0) is a Left
TurnID(Lane #14, Lane #20, Intersection #0) is a Straight
TurnID(Lane #14, Lane #21, Intersection #0) is a Straight
TurnID(Lane #16, Lane #6, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #16, Lane #13, Intersection #0) is a Crosswalk
TurnID(Lane #17, Lane #12, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #17, Lane #22, Intersection #0) is a Crosswalk
TurnID(Lane #18, Lane #3, Intersection #0) is a Left
TurnID(Lane #18, Lane #4, Intersection #0) is a Left
TurnID(Lane #18, Lane #10, Intersection #0) is a Right
TurnID(Lane #18, Lane #11, Intersection #0) is a Right
TurnID(Lane #18, Lane #15, Intersection #0) is a Straight
TurnID(Lane #19, Lane #3, Intersection #0) is a Left
TurnID(Lane #19, Lane #4, Intersection #0) is a Left
TurnID(Lane #19, Lane #10, Intersection #0) is a Right
TurnID(Lane #19, Lane #11, Intersection #0) is a Right
TurnID(Lane #19, Lane #15, Intersection #0) is a Straight
TurnID(Lane #22, Lane #0, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #22, Lane #17, Intersection #0) is a Crosswalk
